pybossa/api/__init__.py

```diff
@@ -391,19 +391,20 @@ def task_progress(project_id=None, short_name=None):
         return abort(404)
 
     sql_text = "SELECT COUNT(*) FROM task WHERE project_id=" + str(project.id)
-
     task_info_fields = get_searchable_columns(project.id)
 
     # create sql query from filter fields received on request.args
     for key in filter_fields.keys():
         if key in task_fields:
             sql_text += " AND {0}=:{1}".format(key, key)
         elif key in task_info_fields:
-            sql_text += " AND info ->> '{0}'=:{1}".format(key, key)
+            if filter_fields[key] == "null":
```

Comment:
Can `null` be case-insensitive (to support `null` and `NULL`)?

Also, we might want to consider re-using the query string for a potential optimization:

```python
sql_text +=  " AND info ->> '{0}'"

if filter_fields[key].lower() == "null":
    sql_text +=  " is Null".format(key)    
else:
    sql_text += "=:{1}".format(key, key)
```

Reply:
Okay, I will change this as well. 
Good idea to use .lower() for null. I'm not sure that I understand how putting the sql_text at the top is an improvement. This implementation will break the .format() as you can see